subprojects/distributions/binary-compatibility.gradle

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+
+import japicmp.model.JApiChangeStatus
+import me.champeau.gradle.japicmp.JapicmpTask
+import org.gradle.api.attributes.Attribute
+import org.gradle.binarycompatibility.rules.*
+import org.gradle.binarycompatibility.transforms.*
+
+buildscript {
+    repositories {
+        maven {
+            url "https://plugins.gradle.org/m2/"
+        }
+    }
+    dependencies {
+        classpath 'me.champeau.gradle:japicmp-gradle-plugin:0.2.1'
+    }
+}
+
+repositories {
+    ivy {
+        name 'Gradle distributions'
+        url 'https://services.gradle.org'
+        layout 'pattern', {
+            ivy '[module]-ivy.xml'
+            artifact '/distributions/[module]-[revision]-bin(.[ext])'
+        }
+    }
+}
+
+configurations {
+    baseline
+}
+
+def ARTIFACT_TYPE = Attribute.of('artifactType', String)
+Set publicProjects = pluginProjects + implementationPluginProjects + publicGroovyProjects - [project(':smokeTest'), project(':soak')]
+def projects = publicProjects.collect { it.archivesBaseName - 'gradle-' }
+def baselineVersion = project.findProperty('baselineVersion')?:releasedVersions.mostRecentFinalRelease
+
+dependencies {
+    baseline "gradle:gradle:${baselineVersion}@zip"
+
+    // This transform takes the Gradle zip distribution,
+    // and unzips the Gradle jar files that it contains in a directory
+    registerTransform {
+        from.attribute(ARTIFACT_TYPE, 'zip')
+        to.attribute(ARTIFACT_TYPE, 'gradle-libs-dir')
+        artifactTransform(ExplodeZipAndFindJars)
+    }
+
+    projects.each { projectName ->
+        // This transform uses the result of the exploded zip extraction
+        // and returns a single jar file based on the lookup project name
+        registerTransform {
+            from.attribute(ARTIFACT_TYPE, 'gradle-libs-dir')
+            to.attribute(ARTIFACT_TYPE, projectName)
+            artifactTransform(FindGradleJar) {
+                params(projectName)
+            }
+        }
+    }
+}
+
+def acceptedViolations = loadBreakingChanges(project, baselineVersion)
+def baselineConfigurations = []
+projects.each { projectName ->
```

Comment:
How do we know that the baseline has the same project names as the current Gradle version?